controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1355,6 +1362,66 @@ func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisio
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) checkServicing(prov provisioner.Provisioner, info *reconcileInfo) (result actionResult, isServicing bool) {
+	servicingData := provisioner.ServicingData{}
+
+	var fwDirty bool
+	if !reflect.DeepEqual(info.host.Status.Provisioning.Firmware, info.host.Spec.Firmware) {
+		servicingData.FirmwareConfig = info.host.Spec.Firmware
+		fwDirty = true
+	}
+
+	hfsDirty, hfs, err := r.getHostFirmwareSettings(info)
+	if err != nil {
+		return actionError{fmt.Errorf("could not determine updated settings: %w", err)}, false
+	}
+	if hfsDirty {
+		servicingData.ActualFirmwareSettings = hfs.Status.Settings
+		servicingData.TargetFirmwareSettings = hfs.Spec.Settings
+	}
+
+	dirty := fwDirty || hfsDirty
+
+	// Even if settings are clean, we need to check the result of the current servicing.
+	if !dirty && info.host.Status.OperationalStatus != metal3api.OperationalStatusServicing && info.host.Status.ErrorType != metal3api.ServicingError {
+		// If nothing is going on, return control to the power management.
+		return nil, false
+	}
+
+	provResult, started, err := prov.Service(servicingData, dirty,
+		info.host.Status.ErrorType == metal3api.ServicingError)
+	if err != nil {
+		return actionError{fmt.Errorf("error servicing host: %w", err)}, false
```

Comment:
Currently this is ignored as if you had returned nil.

Reply:
If I understood correctly, we just need the return here
```
return actionError{fmt.Errorf("error servicing host: %w", err)}
```
Is this a correct assumption @zaneb ?

Reply:
Yep, due to https://github.com/metal3-io/baremetal-operator/pull/1969#discussion_r1806419875 we no longer need to return multiple values from this function at all.

Reply:
TODO(janders): should be resolved in latest commit, make sure it stays this way after I reapply the other fix